nautobot/docs/development/core/ui-component-framework.md

```diff
@@ -0,0 +1,540 @@
+</div>
+<!-- markdownlint-enable no-inline-html -->
+
+## Panel Types
+
+### 0. Base Panel
+
+The Panel component serves as a base class for creating individual display panels within a Layout system.
+
+[Code reference](../../code-reference/nautobot/apps/ui.md#nautobot.apps.ui.Panel)
+
+#### Panel Examples
+
+```python
+from nautobot.apps.ui import Panel, SectionChoices
+
+Panel(
+    label="Panel Header",
+    section=SectionChoices.FULL_WIDTH,
+    weight=100,
+)
+```
+
+```python
+from nautobot.apps.ui import Panel, SectionChoices
+
+Panel(
+    label="Optional Params Included",
+    section=SectionChoices.RIGHT_HALF,
+    weight=200,
+    body_content_template_path="path/to/template/body_content_template.html",
+    header_extra_content_template_path="path/to/template/header_extra_content_template.html",
+    footer_content_template_path="path/to/template/footer_content_template.html",
+    template_path="path/to/template/template.html",
+    body_wrapper_template_path="path/to/template/body_wrapper_template.html",
+)
+```
+
+### 1. ObjectFieldsPanel
+
+`ObjectFieldsPanel` is designed to automatically render object attributes in a table format. It's particularly useful for displaying model instances or any object with defined attributes. This panel inherits from `KeyValueTablePanel`.
+
+[Code reference](../../code-reference/nautobot/apps/ui.md#nautobot.apps.ui.ObjectFieldsPanel)
+
+NOTE:
+    When `fields="__all__"`, the panel automatically excludes: ManyToMany fields, Reverse relations, Hidden fields and Special fields (`id`, `created`, `last_updated`, `comments`, `tags`). `comments` and `tags` are automatically added as standalone panels.
+
+#### ObjectFieldsPanel Examples
+
+```python
+from nautobot.apps.ui import ObjectFieldsPanel, SectionChoices
+
+ObjectFieldsPanel(
+    label="Object Fields Panel",
+    section=SectionChoices.LEFT_HALF,
+    context_object_key="obj",
+)
+```
+
+```python
+from nautobot.apps.ui import ObjectFieldsPanel, SectionChoices
+
+ObjectFieldsPanel(
+    label="Object Fields Panel",
+    fields=["name", "status", "type", "notes"],
+    section=SectionChoices.LEFT_HALF,
+    context_object_key="obj",
+    ignore_nonexistent_fields=True,
+    weight=200,
+    value_transforms={
+        "status": [status_to_badge],
+        "type": [type_to_link]
+    },
+)
+```
+
+<!-- markdownlint-disable no-inline-html -->
+<div class="grid cards" style="width: 300px;" markdown>
+- ![ObjectFieldsPanel Example](../../media/development/core/ui-component-framework/object-fields-panel-example.png){ .on-glb }
+</div>
+<!-- markdownlint-enable no-inline-html -->
+
+### 2. KeyValueTablePanel
+
+`KeyValueTablePanel` is a Panel component that displays data in a two-column table format, commonly used in object detail views. It extends the base `Panel` class and provides additional functionality for data display and transformation.
```

Comment:
```suggestion
`KeyValueTablePanel` is a Panel component that displays data in a two-column table format, commonly used in object detail views. It extends the base `Panel` class and provides additional functionality for data display and transformation.

!!! tip
    For displaying the attributes of a specific data model instance, you'll more commonly want to use the `ObjectFieldsPanel` subclass rather than using a `KeyValueTablePanel` directly.
```